android/KMAPro/kMAPro/src/main/java/com/keyman/android/SystemKeyboard.java

```diff
@@ -166,8 +168,11 @@ public void onKeyboardLoaded(KeyboardType keyboardType) {
   }
 
   @Override
-  public void onKeyboardChanged(String newKeyboard) {
-    // Do nothing
+  public void onKeyboardChanged(String newKeyboard, KeyboardType keyboardType) {
+    // System keyboard changed, so refresh the chiral modifier status
+    if (keyboardType == KeyboardType.KEYBOARD_TYPE_SYSTEM) {
```

Comment:
So I am uncomfortable with this change. This code needs to be in KMEA, not in the app: it's not something that every app developer that uses Keyman Engine should have to worry about.

Reply:
> Finally, is the chirality test stable?
ah, the fun and interesting part of this PR. Afaict, the test works if the keyboard is done loading. But since KMManager changes both in-app and system keyboard, the timing for everything just isn't consistent to call the test stable.

Reply:
So I've attached the chirality checks to KMW event handlers, and the test is still inconsistent. Now, it seems the initial check of a chiral keyboard is wrong, but after toggling to another keyboard and back, it becomes correct. This happens on both in-app and system keyboard.